src/ci/docker/scripts/secure-download.sh

```diff
@@ -0,0 +1,21 @@
+/bin/bash
+
+set -ex
+
+if [[ -z "$2" ]]; then
+  echo "usage: secure-download.sh <URL> <SHA256>"
+  exit 1
+fi
+
+URL=$1
+SHA256=$2
+BASENAME=`basename ${URL}`
+
+ON_EXIT="rm -f ${BASENAME}"
+
+trap "rm -f ${BASENAME}" EXIT
```

Comment:
Wow, so one needs *both* `ON_EXIT` and `trap` to reliably clean up? Shell scripting makes me sad. :(

Might be worth adding a comment with (a link to) some explanation why, if a good one exists.